cdap-app-templates/cdap-program-report/src/main/java/co/cask/cdap/report/ReportGenerationSpark.java

```diff
@@ -466,22 +612,42 @@ private void generateReport(ReportGenerationRequest reportRequest, Location repo
     }
 
     /**
-     * Returns the status of the report generation by checking the presence of the success file or the failure file.
-     * If neither of these files exists, the report generation is still running.
+     * Gets the expiry time in milliseconds of a report by adding the VALID_DURATION_MILLIS
+     * to the report generation completed time read from the _SUCCESS file.
+     *
+     * @param reportIdDir the base directory with report ID as directory name
+     * @return the expiry time in milliseconds or {@code null} if the expiry time cannot be determined
+     * @throws Exception _SUCCESS doesn't exist or reading the report generation completed time from it failed
+     */
+    private static Long getExpiryTimeMillis(Location reportIdDir) throws Exception {
```

Comment:
its better to keep the time unit as seconds. as the caller is converting it to seconds anyway.

Reply:
there's also another usage of this method to compare the time returned directly with `System.currentTimeMillis()`